### DISCUSSION TITLE
[Feature] Migration skywalking from Junit4 to Junit5

### DISCUSSION THREAD

**Discussion by GOODBOY008** [apache/skywalking#8555] [Ideas]
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/skywalking/issues?q=is%3Aissue) and found no similar feature requirement.

### Description

The main motivations are:
- Making full use of Java 8 Lambdas for writing easier to read tests and a
better performing way of composing failure messages.
- Improved test structures with nested and dynamic tests.
- Much better support for parameterized tests to avoid separating
parameterized and non-parameterized parts into different test classes.
- Composable dependencies and better hooks for advanced use cases.
- Better exception verification
- More current infrastructure
- Better parallelizable

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

**Comment by wu-sheng** [comment]
If you want to do this, I am fine. But notice, there are several Java repositories, backend and agent are major.

**Comment by wu-sheng** [comment]
But the work will be huge, and this wouldn't be easy to review.

**Comment by lujiajing1126** [comment]
As I understand, if we are using `powermock`, migrating to `Junit5` may not be possible due to the upstream issue https://github.com/powermock/powermock/issues/929.

**Comment by wu-sheng** [comment]
@lujiajing1126 Thank you for updating. And yes, powermock is common in our UTs. I am moving this to discussion for now.

**Comment by GOODBOY008** [comment]
We can migrate step by step to complete this work,when powermock support Junit5.This also will confuse developer when Junit4 mixed with Junit5.
